frontend/coprs_frontend/coprs/forms.py

```diff
@@ -1329,6 +1329,14 @@ def _validate_batch_opts(form, field):
     return F
 
 
+class BuildFormCheckFactory:
+    """
+    For checking before SRPM upload
+    """
+    def __new__(cls, active_chroots, package=None):
+        return _get_build_form(active_chroots, BasePackageForm, package)
```

Comment:
This makes sense, nice.

Reply:
Even though, shouldn't we use `BaseForm` instead of `BasePackageForm`?  I'm - again after some time - lost in the Form abstraction :-)  forms.py deserves some simplification IMO.

Reply:
> Even though, shouldn't we use BaseForm instead of BasePackageForm? 

You are right, `BaseForm` should work as well. I chose `BasePackageForm` because some (and I thought that all) other build forms inherit from it. But funnily enough, the only exception is `BuildFormUploadFactory`. I'll change this PR to use `BaseForm`.

> I'm - again after some time - lost in the Form abstraction :-) 

As it turns out, so do I :-)

> forms.py deserves some simplification IMO.

I am hoping it will be touched upon during the OpenAPI implementation.

Reply:
Updated